### DISCUSSION TITLE
Using cordova plugins in a capacitor-powered app

### DISCUSSION THREAD

**Discussion by please-fill-out-this-field** [quasarframework/quasar#15265] [CLI - Capacitor / Cordova modes]
Hello!

I have an app that's build around Capacitor v3 and now I need to add some cordova plugins. How do I use them?

The [in-app browser plugin](https://github.com/apache/cordova-plugin-inappbrowser), for instance. I installed the plugin and did that `cap sync` thing - capacitor detected "1 cordova plugin for android", no problems here. The problem is that plugin's documentation tells me to
```js
var ref = cordova.InAppBrowser.open('http://apache.org', '_blank', 'location=yes');
```
but I have no idea where to get the `cordova` object from.

First I tried to just `yarn add cordova` then
```js
// SomeComponent.vue
import cordova from 'cordova'
```
but it won't build even for web - `quasar dev` failed with kilometers of import errors. 
```
App •  ERROR  •  UI  in ./node_modules/http-proxy-agent/dist/agent.js

Module not found: Can't resolve imported dependency "tls"
Did you forget to install it? You can run: yarn add tls

 App •  ERROR  •  UI  in ./node_modules/http-proxy-agent/dist/agent.js

Module not found: Can't resolve imported dependency "url"
Did you forget to install it? You can run: yarn add url

 App •  ERROR  •  UI  in ./node_modules/http-signature/lib/parser.js

Module not found: Can't resolve imported dependency "util"
Did you forget to install it? You can run: yarn add util

# and so and so on...
```

Then I tried to import the plugin directly:
```js
import whatever from 'cordova-plugin-inappbrowser'
```
but apparently cordova plugins don't work that way:
```
 App •  ERROR  •  UI  in ./src/layouts/MainLayout.vue?vue&type=script&lang=js

Module not found: Can't resolve imported dependency "cordova-plugin-inappbrowser"
Did you forget to install it? You can run: yarn add cordova-plugin-inappbrowser


 App •  COMPILATION FAILED  • Please check the log above for details.
```

Well, that's it. Any help is appreciated.

P.S. No, capacitor's browser doesn't satisfy my needs. No, I don't wanna switch to cordova entirely.

**Comment by please-fill-out-this-field** [comment]
Already figured out. Doesn't work in browser though...
Well, thanks anyway.

пт, 13 янв. 2023 г. в 19:58, Aldrin Marquez ***@***.***>:

**Reply by tempire** [reply]
If you've figured it out, post the solution.
